src/codegen/riscv64/assembler-riscv64.cc

```diff
@@ -1798,6 +1809,81 @@ void Assembler::li_constant(Register rd, int64_t imm) {
   addi(rd, rd, imm << 52 >> 52);  // Bits 11:0
 }
 
+int64_t Assembler::decode_li_constant(Address pc) {
+  DCHECK(IsLui(instr_at(pc)));
+  DCHECK(IsAddiw(instr_at(pc + 4)));
+  DCHECK(IsSlli(instr_at(pc + 8)));
+  DCHECK(IsAddi(instr_at(pc + 12)));
+  DCHECK(IsSlli(instr_at(pc + 16)));
+  DCHECK(IsAddi(instr_at(pc + 20)));
+  DCHECK(IsSlli(instr_at(pc + 24)));
+  DCHECK(IsAddi(instr_at(pc + 28)));
+
+  int32_t imm_63_48 =
+      Instruction::At(reinterpret_cast<byte*>(pc))->Imm20UValue() << 12;
+  int32_t imm_47_36 =
+      Instruction::At(reinterpret_cast<byte*>(pc + 4))->Imm12Value();
+  int64_t imm = imm_63_48 + imm_47_36;
+  imm = imm << 12;
+  imm = imm + Instruction::At(reinterpret_cast<byte*>(pc + 12))->Imm12Value();
+  imm = imm << 12;
+  imm = imm + Instruction::At(reinterpret_cast<byte*>(pc + 20))->Imm12Value();
+  imm = imm << 12;
+  imm = imm + Instruction::At(reinterpret_cast<byte*>(pc + 28))->Imm12Value();
+  DEBUG_PRINTF("decode_li_constant 0x%lx\n", imm);
+  return imm;
+}
+
+void Assembler::patch_li_constant(Address pc, int64_t imm) {
```

Comment:
Can we use the existing `set_target_value_at` method instead? Is there anything different?